pkg/gcp/resources.go

```diff
@@ -488,12 +575,150 @@ func (r *gcpGKE) fromResourceDecription(resourceDesc []byte) (*containerpb.Creat
+	return firewallTarget{TargetType: targetTypeAddress, Target: netInfo.Address}
+}
+
+// Get network information about a service attachment
+// Returns the network name, resource ID (service attachment ID, not URL since this is used for firewall rule naming), and IP address
+func (r *privateServiceHandler) getNetworkInfo(ctx context.Context, resourceInfo *resourceInfo) (*resourceNetworkInfo, error) {
+	// Get the service attachment information
+	attachmentRequest := &computepb.GetServiceAttachmentRequest{
+		ServiceAttachment: resourceInfo.Name,
+		Project:           resourceInfo.Project,
+		Region:            resourceInfo.Region,
+	}
+	resp, err := r.attachmentsClient.Get(ctx, attachmentRequest)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get attachment: %w", err)
+	}
+
+	// Get the address information
+	addressRequest := &computepb.GetAddressRequest{
+		Address: getAddressName(resourceInfo.Name),
+		Project: resourceInfo.Project,
+		Region:  resourceInfo.Region,
+	}
+	addr, err := r.addressesClient.Get(ctx, addressRequest)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get address: %w", err)
+	}
+
+	return &resourceNetworkInfo{NetworkName: getVpcName(resourceInfo.Namespace), ResourceID: convertIntIdToString(*resp.Id), Address: *addr.Address}, nil
+}
+
+// Create a private service connect endpoint with given network settings
+func (r *privateServiceHandler) createWithNetwork(ctx context.Context, service ServiceAttachmentDescription, subnetName string, resourceInfo *resourceInfo) (string, string, error) {
+	// Reserve an IP address to be the endpoint
+	addressName := getAddressName(resourceInfo.Name)
+	addrRequest := computepb.InsertAddressRequest{
+		Project: resourceInfo.Project,
+		Region:  resourceInfo.Region,
+		AddressResource: &computepb.Address{
+			Name:        &addressName,
+			Subnetwork:  proto.String(getSubnetworkUrl(resourceInfo.Project, resourceInfo.Region, subnetName)),
+			AddressType: proto.String("INTERNAL"),
+			IpVersion:   proto.String("IPV4"),
+		},
+	}
+	addrOp, err := r.addressesClient.Insert(ctx, &addrRequest)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to insert address: %w", err)
+	}
+	if err = addrOp.Wait(ctx); err != nil {
+		return "", "", fmt.Errorf("unable to wait for the operation: %w", err)
+	}
+
+	// Get the allocated address
+	getAddressReq := computepb.GetAddressRequest{
+		Project: resourceInfo.Project,
+		Region:  resourceInfo.Region,
+		Address: addressName,
+	}
+	addr, err := r.addressesClient.Get(ctx, &getAddressReq)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to get address: %w", err)
+	}
+
+	// Create a forwarding rule for the endpoint
+	forwardingRuleRequest := computepb.InsertForwardingRuleRequest{
+		Project: resourceInfo.Project,
+		Region:  resourceInfo.Region,
+		ForwardingRuleResource: &computepb.ForwardingRule{
+			Name:      proto.String("forwarding-rule-" + resourceInfo.Name),
```

Comment:
Do we need to add the paraglider prefix here? On the same note, is it beneficial to make this into a function like we have for other resource names?